homeassistant/components/bmw_connected_drive/config_flow.py

```diff
@@ -32,19 +32,22 @@ async def validate_input(
 
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
-    account = MyBMWAccount(
+    auth = MyBMWAuthentication(
         data[CONF_USERNAME],
         data[CONF_PASSWORD],
         get_region_from_name(data[CONF_REGION]),
     )
 
     try:
-        await account.get_vehicles()
+        await auth.login()
     except HTTPError as ex:
         raise CannotConnect from ex
 
     # Return info that you want to store in the config entry.
-    return {"title": f"{data[CONF_USERNAME]}{data.get(CONF_SOURCE, '')}"}
+    retval = {"title": f"{data[CONF_USERNAME]}{data.get(CONF_SOURCE, '')}"}
+    if auth.refresh_token:
+        retval[CONF_REFRESH_TOKEN] = auth.refresh_token
```

Comment:
Please add a test which shows the refresh token is stored as intended.

Reply:
Could you help me here a little? I'm struggling to implement the required mock for the test.

`auth.refresh_token` is stored in `auth` by `auth.login()` (but it is not `auth.login()`'s return value). Is there any way to set this during runtime?

Reply:
Instead of patching out `bimmer_connected.api.authentication.MyBMWAuthentication.login`, patch the `httpx` API.

There are quite a lot of tests which do that, I think you can do something like this:
```
    respx.get("http://lcocoapi.bmwgroup.com").respond(
        status_code=HTTPStatus.OK, json={blah}
    )
```

Reply:
Yeah, we're doing this in the library already but didn't want to re-rewite the tests. Will see to get a minimal working set of required mocks.

Reply:
You could also set a side effect on the patched login method that sets the appropriate auth state, tokens etc.